### ISSUE TITLE
How to describe search space like xgb in space.py

### ISSUE THREAD

**Issue by yangzhg** [AIworx-Labs/chocolate#27]
how to describe  xgb parameters inlcude booster subparameters using space.py, 
mycode
 ```
{
            "min_child_weight": choco.uniform(0, 20),
            "n_estimators": choco.quantized_uniform(25, 525, 20),
            "max_depth": choco.quantized_uniform(1, 20, 1),
            "subsample": choco.uniform(0.7, 1.0),
            "learning_rate": choco.uniform(0.001, 1.0),
            "colsample_bytree": choco.uniform(0.1, 1.0),
            "colsample_bylevel": choco.uniform(0.1, 1.0),
            "reg_alpha": choco.log(-10, -1, 10),
            "reg_lambda": choco.log(-10, -1, 10),
            "booster": {
                "gbtree": None,
                "gblinear": {
                    "updater": {
                        "shotgun": None, "coord_descent": None
                    },
                    "feature_selector": {
                        "cyclic": None, "shuffle": None,
                        "random": None, "greedy": None,
                        "thrifty": None
                    }
                },
                "dart": {
                    "sample_type": {
                        "uniform": None,
                        "weighted": None
                    },
                    "normalize_type": {
                        "tree": None,
                        "forest": None
                    },
                    "rate_drop": choco.uniform(0.0, 1.0),
                    "skip_drop": choco.uniform(0.0, 1.0),
                }
            },
        }
```
only generate
 `{'booster': 'dart', 'colsample_bylevel': 0.18988467350254096, 'colsample_bytree': 0.14279195511563256, 'learning_rate': 0.5499628359285743, 'max_depth': 12, 'min_child_weight': 11.517182050898144, 'n_estimators': 65, 'reg_alpha': 0.0008107887387917439, 'reg_lambda': 1.995310903080172e-10, 'subsample': 0.8042911086851237}`,`sample_type` and `normalize_type` dose not generated

**Comment by leconteur** [comment]
In this case, you should use choco.choice(["uniform", "weighted"]) as a distribution instead of a dict containing an empty key.
